### ISSUE TITLE
the performance of tensorflow distributed

### ISSUE THREAD

**Issue by dingevin** [tensorflow/tensorflow#23710]
**Describe the current behavior**
I have trained a speech recognition network using tensorflow both on multi-gpu single machine version(multi-gpu) and multi-gpu multi-machine(distributed) version. the training speed was ok in multi-GPU but the speed on distributed is slow and accuracy is not same as multi-GPU version at same steps. I watched the timeline and found RecvTensor is waste so long time. i‘m wondering how to improve the performance of distributed training.
![c2e456a6 17eda71d 1b93e65b](https://user-images.githubusercontent.com/16797858/48396541-c0f02480-e755-11e8-97fd-8dad532956fa.png)

**log info**
the start script for each worker (3 workers in total and 3 ps) is:
```
CUDA_VISIBLE_DEVICES='' nohup python distributed.py --ps_hosts=gpu47-hca:2220,gpu42-hca:2221,gpu35-hca:2222 --worker_hosts=gpu47-hca:3330,gpu42-hca:3331,gpu35-hca:3332 --job_name=ps --task_index=0 &

nohup python distributed.py --ps_hosts=gpu47-hca:2220,gpu42-hca:2221,gpu35-hca:2222 --worker_hosts=gpu47-hca:3330,gpu42-hca:3331,gpu35-hca:3332 --job_name=worker --task_index=0 --gpu_num=4 --learning_rate=0.00001 --batch_size=128&
```
the training log:
multi-gpu:
```
INFO:tensorflow:Step #312: rate 0.000010, accuracy 0.08653%, cross entropy 7.359044(192.9 examples/sec; 0.166 sec/batch)
INFO:tensorflow:Step #313: rate 0.000010, accuracy 0.09099%, cross entropy 7.384690(218.1 examples/sec; 0.147 sec/batch)
INFO:tensorflow:Step #314: rate 0.000010, accuracy 0.09412%, cross entropy 7.325777(241.5 examples/sec; 0.132 sec/batch)
INFO:tensorflow:Step #315: rate 0.000010, accuracy 0.08608%, cross entropy 7.381987(225.8 examples/sec; 0.142 sec/batch)
```
distributed:
```
INFO:tensorflow:Step #768: learning_rate:0.00001, accuracy 0.49589%, cross entropy 6.968968 (62.8 examples/sec; 0.509 sec/batch)
INFO:tensorflow:Step #769: learning_rate:0.00001, accuracy 0.60303%, cross entropy 7.082638 (87.0 examples/sec; 0.368 sec/batch)
INFO:tensorflow:Step #770: learning_rate:0.00001, accuracy 0.77971%, cross entropy 7.041758 (54.9 examples/sec; 0.583 sec/batch)
INFO:tensorflow:Step #771: learning_rate:0.00001, accuracy 1.00002%, cross entropy 6.996723 (41.4 examples/sec; 0.773 sec/batch)
```

**System information**
- Have I written custom code: yes    (code is [here](https://github.com/dingevin/distributed-training/blob/master/distributed.py))
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): CentOS release 6.8
- TensorFlow installed from (source or binary): binary 
- TensorFlow version (use command below): 1.10.1
- Python version: 2.7.13
- GPU model and memory: TITAN X 12G
- Network bandwidt: 125M/s

**What I have tried**
1. the **RecvTensor** is so long time , so i increase the number of ps, the speed more fast than before but accuracy don't improve. i wonder how to confirm the number of ps and worker？if i increate the numer of ps the speed can be faster?  my network is 1CNN+5LSTM+1Fully Connected Layer。
2. the accuracy was 10% in multi-gpu 4000 steps; so i set 2000 steps to each worer(2 ps 2 worker), but the accuracy is 5%， it looks don't sync updata parameter? Is there something wrong with my [code](https://github.com/dingevin/distributed-training/blob/master/distributed.py)?

**Comment by wt-huang** [comment]
The accuracy is very low, the learning rate is also low. 
What is the accuracy number on a simple model using your setup?

**Comment by dingevin** [comment]
@wt-huang ,the accuracy can be reach 80% if training long time at multi-gpu. the accuracy is slow because only training a little steps and low learning rate; my questions is:
1. when i training same steps on multi-gpu  and distributed, the accuracy not same approximately；i mean if training 4000 steps on multi-gpu, the accuracy can reach 10%， but the accuracy was 5% at distributed each worker training 2000 steps, it looks that  my code didn't share the parameters.
2. and the training speeds is slow than multi-gpu， i don't know why.
